package/src/main.rs

```diff
@@ -27,26 +31,116 @@ const CARGO_MANIFEST_DIR: &str = env!("CARGO_MANIFEST_DIR");
 const OUT_DIR: &str = env!("OUT_DIR");
 const TARGET: &str = env!("TARGET");
 
-#[cfg(target_family = "windows")]
+#[cfg(windows)]
 const PATHSEP: &str = ";";
 
-#[cfg(target_family = "unix")]
+#[cfg(windows)]
+const EOL: &str = "\r\n";
+
+#[cfg(unix)]
 const PATHSEP: &str = ":";
 
+#[cfg(unix)]
+const EOL: &str = "\n";
+
+#[derive(Eq, PartialEq)]
+enum Platform {
+    LinuxAarch64,
+    LinuxX86_64,
+    MacOSAarch64,
+    MacOSX86_64,
+    WindowsX86_64,
+}
+
+impl Platform {
+    fn current() -> Result<Self, Exit> {
+        match (env::consts::OS, env::consts::ARCH) {
+            ("linux", "aarch64") => Ok(Self::LinuxAarch64),
+            ("linux", "x86_64") => Ok(Self::LinuxX86_64),
+            ("macos", "aarch64") => Ok(Self::MacOSAarch64),
+            ("macos", "x86_64") => Ok(Self::MacOSX86_64),
+            ("windows", "x86_64") => Ok(Self::WindowsX86_64),
+            _ => Err(Code::FAILURE.with_message({
+                format!(
+                    "Unsupported platform: os={os} arch={arch}",
+                    os = env::consts::OS,
+                    arch = env::consts::ARCH
+                )
+            })),
+        }
+    }
+
+    fn to_str(&self) -> &str {
+        match self {
+            Platform::LinuxAarch64 => "linux-aarch64",
+            Platform::LinuxX86_64 => "linux-x86_64",
+            Platform::MacOSAarch64 => "macos-aarch64",
+            Platform::MacOSX86_64 => "macos-x86_64",
+            Platform::WindowsX86_64 => "windows-x86_64",
+        }
+    }
+}
+
+impl Display for Platform {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        f.write_str(self.to_str())
+    }
+}
+
 lazy_static! {
-    static ref OS_ARCH: String = format!(
-        "{os}-{arch}",
-        os = env::consts::OS,
-        arch = env::consts::ARCH
-    );
+    static ref CURRENT_PLATFORM: Platform = Platform::current().unwrap();
+}
+
+macro_rules! log {
```

Comment:
Since dependency count doesn't seem to be an issue for this binary, I'm surprised that this isn't using an existing logging package (I could understand wanting to keep the final product thinner though).

Reply:
Well, these are all INFO and I differentiate build steps from ITs with Cyan and Magenta. Certainly dumb, but I don't know how to do this with the logging package and this was easier to figure out. The driver behind this apparent ~frivolity was me myself not understanding what was going on. It's a complex build, and RUST_LOG=debug|trace is useful but ~insane and only if you're in the weeds.

Reply:
I did find simpler color printing crates, but they had ~1k uses. I've been trying to stick to Tolnays, BurntSushis, Chrichton's, etc.